extension/src/debugger/languages/dotnet.ts

```diff
@@ -287,7 +287,18 @@ export function createProjectDebuggerExtension(dotNetServiceProducer: (debugSess
             debugConfiguration.args = determineArguments(baseProfile?.commandLineArgs, args);
             debugConfiguration.executablePath = baseProfile?.executablePath;
             debugConfiguration.checkForDevCert = baseProfile?.useSSL;
-            debugConfiguration.serverReadyAction = determineServerReadyAction(baseProfile?.launchBrowser, baseProfile?.applicationUrl);
+
+            if (launchOptions.isApphost) {
+                // The Aspire dashboard URL will be set as the apphost's application URL. Check if auto-launch is enabled
+                const enableDashboardAutoLaunch = vscode.workspace.getConfiguration('aspire').get<boolean>('enableAspireDashboardAutoLaunch', true);
+
+                if (enableDashboardAutoLaunch) {
+                    debugConfiguration.serverReadyAction = determineServerReadyAction(baseProfile?.launchBrowser, baseProfile?.applicationUrl);
```

Comment:
(nit) is there a way to refactor the logic so we only need to set `debugConfiguration.serverReadyAction` in a single spot?